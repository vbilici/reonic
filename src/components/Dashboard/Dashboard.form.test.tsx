import { render, fireEvent } from '@testing-library/react';
import ConfigForm from './Dashboard.form';

describe('ConfigForm', () => {
  const mockOnSubmit = jest.fn();
  it('renders the form correctly', () => {
    const { getByText } = render(<ConfigForm onSubmit={mockOnSubmit} />);

    expect(getByText('Number of Chargepoints:')).toBeInTheDocument();

    expect(getByText('Charging Power per Chargepoint (Kw):')).toBeInTheDocument();

    expect(getByText('Consumption of Cars:')).toBeInTheDocument();

    expect(getByText('Run The Simulation')).toBeInTheDocument();
  });

  it('calls onSubmit with the correct form state when submitted', () => {
    const { getByText, getByTestId } = render(<ConfigForm onSubmit={mockOnSubmit} />);

    const numberOfChargepointsInput = getByTestId('numberOfChargepoints');
    const chargingPowerInput = getByTestId('chargingPowerPerChargepoint');
    const consumptionOfCarsInput = getByTestId('consumptionOfCars');

    fireEvent.change(numberOfChargepointsInput, { target: { value: '30' } });
    fireEvent.change(chargingPowerInput, { target: { value: '15' } });
    fireEvent.change(consumptionOfCarsInput, { target: { value: '20' } });

    fireEvent.click(getByText('Run The Simulation'));

    expect(mockOnSubmit).toHaveBeenCalledWith({
      numberOfChargepoints: 30,
      chargingPowerPerChargepoint: 15,
      consumptionOfCars: 20,
      arrivalProbabilityMultiplier: 100
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ConfigForm onSubmit={mockOnSubmit} />);
    expect(asFragment()).toMatchSnapshot();
  });
});