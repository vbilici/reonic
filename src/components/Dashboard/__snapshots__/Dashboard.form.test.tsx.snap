// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="tremor-Card-root relative w-full text-left ring-1 rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand mb-6"
  >
    <form>
      <div
        class=""
      >
        <div
          class="flex flex-wrap gap-4 items-center mb-4"
        >
          <p
            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content"
          >
            Number of Chargepoints:
          </p>
          <div
            class="tremor-NumberInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 border shadow-tremor-input dark:shadow-dark-tremor-input bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis border-tremor-border dark:border-dark-tremor-border ml-2 w-20"
          >
            <input
              class="tremor-NumberInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 py-2 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
              data-testid="numberOfChargepoints"
              name="numberOfChargepoints"
              placeholder="Type..."
              type="number"
              value="20"
            />
            <div
              class="flex justify-center align-middle"
            >
              <div
                class="cursor-pointer hover:text-tremor-content dark:hover:text-dark-tremor-content flex mx-auto text-tremor-content-subtle dark:text-dark-tremor-content-subtle group py-[10px] px-2.5 border-l border-tremor-border dark:border-dark-tremor-border"
                tabindex="-1"
              >
                <svg
                  class=" h-4 w-4 duration-75 transition group-active:scale-95"
                  data-testid="step-down"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 12H4"
                  />
                </svg>
              </div>
              <div
                class="cursor-pointer hover:text-tremor-content dark:hover:text-dark-tremor-content flex mx-auto text-tremor-content-subtle dark:text-dark-tremor-content-subtle group py-[10px] px-2.5 border-l border-tremor-border dark:border-dark-tremor-border"
                tabindex="-1"
              >
                <svg
                  class=" h-4 w-4 duration-75 transition group-active:scale-95"
                  data-testid="step-up"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 4v16m8-8H4"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-wrap gap-4 items-center mb-4"
        >
          <p
            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content"
          >
            Charging Power per Chargepoint (Kw):
          </p>
          <div
            class="tremor-NumberInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 border shadow-tremor-input dark:shadow-dark-tremor-input bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis border-tremor-border dark:border-dark-tremor-border ml-2 w-20"
          >
            <input
              class="tremor-NumberInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 py-2 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
              data-testid="chargingPowerPerChargepoint"
              name="chargingPowerPerChargepoint"
              placeholder="Type..."
              type="number"
              value="11"
            />
            <div
              class="flex justify-center align-middle"
            >
              <div
                class="cursor-pointer hover:text-tremor-content dark:hover:text-dark-tremor-content flex mx-auto text-tremor-content-subtle dark:text-dark-tremor-content-subtle group py-[10px] px-2.5 border-l border-tremor-border dark:border-dark-tremor-border"
                tabindex="-1"
              >
                <svg
                  class=" h-4 w-4 duration-75 transition group-active:scale-95"
                  data-testid="step-down"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 12H4"
                  />
                </svg>
              </div>
              <div
                class="cursor-pointer hover:text-tremor-content dark:hover:text-dark-tremor-content flex mx-auto text-tremor-content-subtle dark:text-dark-tremor-content-subtle group py-[10px] px-2.5 border-l border-tremor-border dark:border-dark-tremor-border"
                tabindex="-1"
              >
                <svg
                  class=" h-4 w-4 duration-75 transition group-active:scale-95"
                  data-testid="step-up"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 4v16m8-8H4"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-wrap gap-4 items-center mb-4"
        >
          <p
            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content"
          >
            Consumption of Cars:
          </p>
          <div
            class="tremor-NumberInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 border shadow-tremor-input dark:shadow-dark-tremor-input bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis border-tremor-border dark:border-dark-tremor-border ml-2 w-20"
          >
            <input
              class="tremor-NumberInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 py-2 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
              data-testid="consumptionOfCars"
              name="consumptionOfCars"
              placeholder="Type..."
              type="number"
              value="18"
            />
            <div
              class="flex justify-center align-middle"
            >
              <div
                class="cursor-pointer hover:text-tremor-content dark:hover:text-dark-tremor-content flex mx-auto text-tremor-content-subtle dark:text-dark-tremor-content-subtle group py-[10px] px-2.5 border-l border-tremor-border dark:border-dark-tremor-border"
                tabindex="-1"
              >
                <svg
                  class=" h-4 w-4 duration-75 transition group-active:scale-95"
                  data-testid="step-down"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 12H4"
                  />
                </svg>
              </div>
              <div
                class="cursor-pointer hover:text-tremor-content dark:hover:text-dark-tremor-content flex mx-auto text-tremor-content-subtle dark:text-dark-tremor-content-subtle group py-[10px] px-2.5 border-l border-tremor-border dark:border-dark-tremor-border"
                tabindex="-1"
              >
                <svg
                  class=" h-4 w-4 duration-75 transition group-active:scale-95"
                  data-testid="step-up"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 4v16m8-8H4"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-row gap-4 items-center"
        >
          <button
            class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm bg-tremor-brand dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis col-start-2 w-40 ml-2"
            type="submit"
          >
            <span
              class="tremor-Button-text text-tremor-default whitespace-nowrap"
            >
              Run The Simulation
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
